tests/test_util_fns.py

```diff
@@ -289,16 +298,82 @@ def test_int_property_sendcmd_query(mock_inst):
     mock_inst.spy_sendcmd.assert_called()
 
 
-def test_unitful_property_sendcmd_query(mock_inst):
-    """Assert that unitful_property calls sendcmd, query of parent class."""
-    # getter
-    assert mock_inst.unitful_property == u.Quantity(42, u.K)
-    mock_inst.spy_query.assert_called()
-    # setter
-    value = 13
-    mock_inst.unitful_property = u.Quantity(value, u.K)
-    assert mock_inst._sendcmd == f"42 {value:e}"
-    mock_inst.spy_sendcmd.assert_called()
+class Test_unitful_property:
+    def test_unitful_property_sendcmd_query(self, mock_inst):
+        """Assert that unitful_property calls sendcmd, query of parent class."""
+        # getter
+        assert mock_inst.unitful_property == u.Quantity(42, u.m)
+        mock_inst.spy_query.assert_called()
+        # setter
+        value = 13
+        mock_inst.unitful_property = u.Quantity(value, u.m)
+        assert mock_inst._sendcmd == f"42 {value:e}"
+        mock_inst.spy_sendcmd.assert_called()
+
+    def test_unitful_property_sendcmd_query_unitless(self, mock_inst):
+        """Assert that unitful_property calls sendcmd, query of parent class.
+        Here for a unitless input
+        """
+        # getter
+        assert mock_inst.unitful_property == u.Quantity(42, u.m)
+        mock_inst.spy_query.assert_called()
+        # setter
+        value = 13
+        mock_inst.unitful_property = value
+        assert mock_inst._sendcmd == f"42 {value:e}"
+        mock_inst.spy_sendcmd.assert_called()
+
+    @pytest.mark.parametrize("value", (0.1, 200, 0.1 * u.m, 200 * u.m))
+    def test_unitful_property_sendcmd_limited_unfit(self, mock_inst, value):
+        """Assert that unitful_property calls sendcmd, query of parent class.
+        Here an input out of bounds for quantity limited property."""
+        # setter
+        with pytest.raises(ValueError):
+            mock_inst.unitful_property_limited = value
+
+    @pytest.mark.parametrize("value", (13 * u.m, 17 * u.m, 55 * u.m))
+    def test_unitful_property_sendcmd_limited_pass_un(self, mock_inst, value):
+        """Assert that unitful_property calls sendcmd, query of parent class.
```

Comment:
Code looks good, but the test docstrings are mostly copy & paste 😄

Reply:
The tests do basically the same, just for different input, therefore I kept the first sentence the same and changed what is specific for the input (numer, quantity, passing the limits, not passing. Whether the limits are given as number or quantity).

Reply:
Oh you're right, my eyes just glossed over that, sorry!